firefox-ios/Client/Coordinators/SummarizeCoordinator.swift

```diff
@@ -0,0 +1,67 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import SummarizeKit
+import Common
+import ComponentLibrary
+import UIKit
+
+/// Conforming types can show and hide the browser content together with its toolbars.
+protocol BrowserContentHiding: AnyObject {
+    func showBrowserContent()
+
+    func hideBrowserContent()
+}
+
+class SummarizeCoordinator: BaseCoordinator {
+    private let browserSnapshot: UIImage
+    private let browserSnapshotTopOffset: CGFloat
+    private weak var browserContentHiding: BrowserContentHiding?
+    private weak var parentCoordinatorDelegate: ParentCoordinatorDelegate?
+    private let windowUUID: WindowUUID
+
+    init(
+        browserSnapshot: UIImage,
+        browserSnapshotTopOffset: CGFloat,
+        browserContentHiding: BrowserContentHiding,
+        parentCoordinatorDelegate: ParentCoordinatorDelegate?,
+        windowUUID: WindowUUID,
+        router: Router
+    ) {
+        self.browserSnapshot = browserSnapshot
+        self.browserSnapshotTopOffset = browserSnapshotTopOffset
+        self.parentCoordinatorDelegate = parentCoordinatorDelegate
+        self.browserContentHiding = browserContentHiding
+        self.windowUUID = windowUUID
+        super.init(router: router)
+    }
+
+    func start() {
+        let model = SummarizeViewModel(
+            loadingLabel: .Summarizer.LoadingLabel,
+            loadingA11yLabel: .Summarizer.LoadingAccessibilityLabel,
+            loadingA11yId: AccessibilityIdentifiers.Summarizer.loadingLabel,
+            summarizeTextViewA11yLabel: .Summarizer.SummaryTextAccessibilityLabel,
+            summarizeTextViewA11yId: AccessibilityIdentifiers.Summarizer.summaryTextView,
+            closeButtonModel: CloseButtonViewModel(
+                a11yLabel: .Summarizer.CloseButtonAccessibilityLabel,
+                a11yIdentifier: AccessibilityIdentifiers.Summarizer.closeButton
+            ),
+            tabSnapshot: browserSnapshot,
+            tabSnapshotTopOffset: browserSnapshotTopOffset
+        ) { [weak self] in
+            guard let self else { return }
+            parentCoordinatorDelegate?.didFinish(from: self)
+            browserContentHiding?.showBrowserContent()
+        } onShouldShowTabSnapshot: { [weak self] in
+            self?.browserContentHiding?.hideBrowserContent()
```

Comment:
@thatswinnie do you have context on the browserContentHiding / Showing code and why we need it? 

Otherwise can chat with @FilippoZazzeroni when he get back